nilearn/_utils/niimg_conversions.py

```diff
@@ -329,6 +352,9 @@ def check_niimg(
     # Otherwise, it should be a filename or a SpatialImage, we load it
     niimg = load_niimg(niimg, dtype=dtype)
 
+    if safe_get_data(niimg).size == 0:
+        raise ValueError("The image is empty.")
```

Comment:
commenting this part definitely sped things up

will open a PR to remove this so we can see if things get better